internal/api/server.go

```diff
@@ -232,11 +249,71 @@ func (s *Server) ReshareVault(c echo.Context) error {
 		asynq.Retention(10*time.Minute),
 		asynq.Queue(tasks.QUEUE_NAME))
 	if err != nil {
+		s.logger.Errorf("ReshareVault: Failed to enqueue task: %v", err)
 		return fmt.Errorf("fail to enqueue task, err: %w", err)
 	}
+
+	// Notify plugin server asynchronously
+	go func() {
```

Comment:
instead of using a go i think it will be better we just make a RPC call directly.
1. RPC call to plugin server , if fail , directly return to client
2. If 1 succeed , then queue the task to redis , and trigger verifier to join 
3. and then return to client 

Currently , using a go routine , it is fire and forget , even Plugin is offline , the front end has no way to know

Reply:
made the fix